app/tests/unit/forms/submission/controller.spec.js

```diff
@@ -99,26 +111,48 @@ describe('templateUploadAndRender', () => {
       id: '123',
     },
     submission: {
+      confirmationId: confirmationId,
+      id: submissionId,
       submission: {
-        submission: {
-          data: {
-            simpletextfield: 'fistName lastName',
-            submit: true,
-          },
+        data: {
+          simpletextfield: 'firstName lastName',
+          submit: true,
         },
       },
     },
   };
 
-  mockCdogsResponse.headers['content-disposition'] = 'attachment; filename=template_hello_world.pdf';
-
   it('should call cdogs service if a body specified', async () => {
     service.read = jest.fn().mockReturnValue(mockTemplateReadResponse);
     cdogsService.templateUploadAndRender = jest.fn().mockReturnValue(mockCdogsResponse);
-    await controller.templateUploadAndRender(templateReq, {}, jest.fn());
+    const { res } = getMockRes();
+
+    await controller.templateUploadAndRender(templateReq, res, jest.fn());
+
+    expect(cdogsService.templateUploadAndRender).toHaveBeenCalledTimes(1);
+    expect(cdogsService.templateUploadAndRender).toHaveBeenCalledWith(templateReq.body);
+    expect(res.set).toHaveBeenCalledWith(
+      expect.objectContaining({
+        'Content-Disposition': 'attachment',
+      })
+    );
+  });
+
+  it('should use the cdogs content disposition when it exists', async () => {
```

Comment:
Added a second test to cover the branch where the content disposition is set in the response.